lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -92,38 +92,84 @@ export class PublicClientApplication {
     // #region Redirect Flow
 
     /**
-     * Set the callback functions for the redirect flow to send back the success or error object.
-     * IMPORTANT: Please do not use this function when using the popup APIs, as it will break the response handling
+     * Set the callback functions for the redirect flow to send back the success or error object, and process
+     * any redirect-related data.
+     * IMPORTANT: Please do not use this function when using the popup APIs, as it may break the response handling
      * in the main window.
      * 
      * @param {@link (AuthCallback:type)} authCallback - Callback which contains
      * an AuthError object, containing error data from either the server
      * or the library, depending on the origin of the error, or the AuthResponse object 
      * containing data from the server (returned with a null or non-blocking error).
      */
-    async handleRedirectCallback(authCallback: AuthCallback): Promise<void> {
+    async handleRedirectWithCallback(authCallback: AuthCallback): Promise<void> {
         // Check whether callback object was passed.
         if (!authCallback) {
             throw BrowserConfigurationAuthError.createInvalidCallbackObjectError(authCallback);
         }
 
         // Set the callback object.
         this.authCallback = authCallback;
+
+        // Check if we need to navigate, otherwise handle hash
+        try {
+            await this.handleRedirectResponse();
+        } catch (err) {
+            this.authCallback(err);
+        }
+    }
+
+    /**
+     * Checks if navigateToLoginRequestUrl is set, and:
+     * - if true, performs logic to cache and navigate 
+     * - if false, handles hash string and parses response
+     */
+    private async handleRedirectResponse(): Promise<void> {
```

Comment:
general comment: Are we back porting these to `msal-core`?

Reply:
Adding @tnorling

Reply:
Yes, I'm looking into a related issue in msal-core so some of these changes may be ported over but I'm not able to say which ones yet